Mathlib/Algebra/GroupPower/Lemmas.lean

```diff
@@ -2,35 +2,1316 @@
+theorem self_cast_nat_mul_cast_nat_mul (m n : ℕ) : _root_.Commute (m * a : R) (n * a : R) :=
+  (Commute.refl a).cast_nat_mul_cast_nat_mul m n
+#align commute.self_cast_nat_mul_cast_nat_mul Commute.self_cast_nat_mul_cast_nat_mul
+
+end
+
+variable [Monoid M] [Group G] [Ring R]
+
+@[simp, to_additive]
+theorem units_zpow_right {a : M} {u : Mˣ} (h : _root_.Commute a u)
+    (m : ℤ) : _root_.Commute a ↑(u ^ m) :=
+  SemiconjBy.units_zpow_right h m
+#align commute.units_zpow_right Commute.units_zpow_right
+#align add_commute.add_units_zsmul_right AddCommute.addUnits_zsmul_right
+
+@[simp, to_additive]
+theorem units_zpow_left {u : Mˣ} {a : M} (h : _root_.Commute (↑u) a)
+  (m : ℤ) : _root_.Commute (↑(u ^ m)) a :=
+  (h.symm.units_zpow_right m).symm
+#align commute.units_zpow_left Commute.units_zpow_left
+#align add_commute.add_units_zsmul_left AddCommute.addUnits_zsmul_left
+
+variable {a b : R}
+
+@[simp]
+theorem cast_int_mul_right (h : _root_.Commute a b) (m : ℤ) : _root_.Commute a (m * b : R) :=
+  SemiconjBy.cast_int_mul_right h m
+#align commute.cast_int_mul_right Commute.cast_int_mul_right
+
+@[simp]
+theorem cast_int_mul_left (h : _root_.Commute a b) (m : ℤ) :
+   _root_.Commute ((m : R) * a) b :=
+  SemiconjBy.cast_int_mul_left h m
+#align commute.cast_int_mul_left Commute.cast_int_mul_left
+
+theorem cast_int_mul_cast_int_mul (h : _root_.Commute a b)
+  (m n : ℤ) : _root_.Commute (m * a : R) (n * b : R) :=
+  SemiconjBy.cast_int_mul_cast_int_mul h m n
+#align commute.cast_int_mul_cast_int_mul Commute.cast_int_mul_cast_int_mul
+
+variable (a) (m n : ℤ)
+
+@[simp]
+theorem cast_int_left : _root_.Commute (m : R) a := by
+  rw [← mul_one (m : R)]
+  exact (one_left a).cast_int_mul_left m
+#align commute.cast_int_left Commute.cast_int_left
+
+@[simp]
+theorem cast_int_right : _root_.Commute a m := by
+  rw [← mul_one (m : R)]
+  exact (one_right a).cast_int_mul_right m
+#align commute.cast_int_right Commute.cast_int_right
+
+/- Porting note: `simp` attribute removed as linter reports:
+simp can prove this:
+  by simp only [Commute.cast_int_right, Commute.refl, Commute.mul_right]
+-/
+-- @[simp]
+theorem self_cast_int_mul : _root_.Commute a (n * a : R) :=
+  (Commute.refl a).cast_int_mul_right n
+#align commute.self_cast_int_mul Commute.self_cast_int_mul
+
+/- Porting note: `simp` attribute removed as linter reports:
+simp can prove this:
+  by simp only [Commute.cast_int_left, Commute.refl, Commute.mul_left]
+-/
+-- @[simp]
+theorem cast_int_mul_self : _root_.Commute ((n : R) * a) a :=
+  (Commute.refl a).cast_int_mul_left n
+#align commute.cast_int_mul_self Commute.cast_int_mul_self
+
+theorem self_cast_int_mul_cast_int_mul : _root_.Commute (m * a : R) (n * a : R) :=
+  (Commute.refl a).cast_int_mul_cast_int_mul m n
+#align commute.self_cast_int_mul_cast_int_mul Commute.self_cast_int_mul_cast_int_mul
+
+end Commute
+
+section Multiplicative
+
+open Multiplicative
+
+-- Porting note: the proof became a little roundabout while porting.
```

Comment:
It's because you have `Nat` open. I can fix it!

Reply:
Used your fix. Thanks.